### PR TITLE
Memoization

### PR DESCRIPTION
**Pull request by cknox404**
Memoize findParent and getEpicLinkField functions

### CODE REVIEW

File: test/jira-connection-test.js [hunk 1]
```diff
@@ -8,4 +8,38 @@ describe('JIRA Connection Tests', function() {
         testJira.projectName.should.eql('test');
       });
   });
+
+  describe('findIssue Memoization', () => {
+    let jiraThing;
```
Comment (mtscout6) [inline test/jira-connection-test.js]: `jiraThing`? Can this get a better name?

File: test/jira-connection-test.js [hunk 2]
```diff
@@ -8,4 +8,38 @@ describe('JIRA Connection Tests', function() {
...
+        });
+    });
+
+    it('findIssue with Different Keys is Run Twice', () => {
+      return jiraThing.findIssue('WHP-9990')
+        .then(() => jiraThing.findIssue('WHP-9991'))
+        .then(() => {
+          assert.equal(spy.calledTwice, true);
+        });
+    });
```
Comment (mtscout6) [inline test/jira-connection-test.js]: Neither of these tests ensure the the same result is returned.

File: test/jira-operations-test.js [hunk 3]
```diff
@@ -79,4 +81,29 @@ describe('JIRA Operations Tests', function() {
...
+    it('listFields with Same JIRA Host is Called Only Once', () => {
+      spy = sinon.spy(dummyJira, 'listFields');
+      dummyJira.host = 'jira.host4.com';
+
+      return getEpicLinkField(dummyJira)
+        .then(() => getEpicLinkField(dummyJira))
+        .then(() =>{
+          assert.equal(spy.calledOnce, true);
+        });
+    });
```
Comment (mtscout6) [inline test/jira-operations-test.js]: No assertions against returned value.

Comment (cknox404) [inline test/jira-operations-test.js]: We are already testing against the returned value using findParent in jira-operations-test.js

Comment (mtscout6) [inline test/jira-operations-test.js]: And the same value is returned when called twice?

File: test/jira-connection-test.js [hunk 4]
```diff
@@ -8,4 +9,50 @@ describe('JIRA Connection Tests', function() {
...
+      let firstResult;
+      return jiraApi.findIssue('WHP-9999')
+        .then(result => {
+          firstResult = result;
+          return jiraApi.findIssue('WHP-9999');
+        })
+        .then(result => {
+          assert.equal(spy.calledOnce, true);
+          assert.equal(result, firstResult);
+        });
```
Comment (mtscout6) [inline test/jira-connection-test.js]: A cleaner way to do this:

```
return Promise.all([
    jiraApi.findIssue('WHP-9999'),
    jiraApi.findIssue('WHP-9999')
  ])
  .then(([first, second]) => {
    assert.equal(spy.calledOnce, true);
    assert.equal(result, firstResult);
  });
```

Utilize the array destructuring support in ES6.

Comment (mtscout6) [inline test/jira-connection-test.js]: Updated the code sorry that last example was off a little bit.

File: test/jira-connection-test.js [hunk 5]
```diff
@@ -8,4 +9,50 @@ describe('JIRA Connection Tests', function() {
...
+    it('findIssue with Different Keys is Run Twice', () => {
+      return jiraApi.findIssue('WHP-9991')
+        .then(result => {
+          assert.equal(result.key, 'WHP-9991');
+          return jiraApi.findIssue('WHP-9992');
+        })
+        .then(result => {
+          assert.equal(spy.calledTwice, true);
+          assert.equal(result.key, 'WHP-9992');
+        });
```
Comment (mtscout6) [inline test/jira-connection-test.js]: Similar cleanup as mentioned above

**Comment by mtscout6** [comment]
Once you are done can you squash all these commits?

File: test/jira-operations-test.js (hunk 3 above)
Comment (mtscout6) [inline test/jira-operations-test.js]: Still don't see this addressed. You did in the jira-connection suite but not here.

Comment (cknox404) [inline test/jira-operations-test.js]: I forgot about those ones. I'll get right on them.

**Comment by mtscout6** [comment]
LGTM

**Pull request merged**